types/node/readline/promises.d.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * The `readline/promise` module provides an API for reading lines of input from a Readable stream one line at a time.
+ *
+ * @see [source](https://github.com/nodejs/node/blob/v18.0.0/lib/readline/promises.js)
+ * @since v17.0.0
+ */
+declare module 'readline/promises' {
+    import { Interface as _Interface, ReadLineOptions, Completer, AsyncCompleter, Direction } from 'node:readline';
```

Comment:
I would prefer if we exported these from a different file and imported them in both readline and readline/promises

Reply:
There doesn't seem to be a good way to make that happen, since using a file like `./readline/common` will result in us having to declare a user-facing module that doesn't exist in order to get around ts(2439). I'm keeping my pathfinder changes in a stash if I missed an idea, however.

Reply:
we promote (advice) sticking to file structure re-assembling native module. The pattern here, @nicolas377 used, is same as used in other place to implement 'promises' (async) version of the api